### PR TITLE
Stabilize the schema file formats

### LINKED ISSUE #3844: Stabilize the schema v1.0.0 and v1.1.0 file formats
Both file formats are listed as experimental:

- [v1.0.0](https://github.com/open-telemetry/opentelemetry-specification/blob/703a0af47acac67775f42b9c86570eebb9283b21/specification/schemas/file_format_v1.0.0.md)
- [v1.1.0](https://github.com/open-telemetry/opentelemetry-specification/blob/703a0af47acac67775f42b9c86570eebb9283b21/specification/schemas/file_format_v1.1.0.md)

Neither has changed in a significant way since their introduction two years ago and there is significant adoption (of the v1.1.0 particularly) in the [semantic conventions](https://github.com/open-telemetry/semantic-conventions/tree/fa0725719bcfbab57fa17961faef5f9448d20828/schemas).

At this point it seems these file formats have de facto become stable given that introducing a breaking change to them would severely affect all semantic convention schema. Additional, given the versioning design baked into the file formats, it would follow that any major changes to these file formats would be released as a new verions.

This stabilization is needed for the Go SIG to also release a stable implementation of our [schema parsing package](https://pkg.go.dev/go.opentelemetry.io/otel/schema) (and from there support resource merge semantics).

### PR DESCRIPTION
**Pull request by MrAlias**
Resolve #3844

Both of these files has not changed in two years and are broadly adopted in our semantic conventions. Given they are effectively stabilized as there is likely no desire to change these in any breaking way, make it official by stabilizing both.

### CODE REVIEW

**Comment by tigrannajaryan** [comment]
We are expecting significant new developments on the Schemas from this OTEP: https://github.com/open-telemetry/oteps/pull/243

I think we need to evaluate the current schema file format from this OTEP's perspective before we mark format Stable.

My strong preference would be that any changes to the Schema file format that come from the OTEP are done in backward compatible way, by introducing new format versions. At the moment I don't see why it should not be possible.

@lquerel @jmacd @jsuereth any thoughts?

**Comment by MrAlias** [comment]
> My strong preference would be that any changes to the Schema file format that come from the OTEP are done in backward compatible way, by introducing new format versions.

Agreed. I think this is true in a more general sense too. Any change submitted to these file formats needs to backwards compatible at this point given their wide adoption already in the semantic conventions.

I think the only question in resolving that OTEP is going to be if the new file format will be a minor or major version bump.

I don't think that will block stabilization of these existing file formats though.

**Comment by lquerel** [comment]
> We are expecting significant new developments on the Schemas from this OTEP: [open-telemetry/oteps#243](https://github.com/open-telemetry/oteps/pull/243)
> 
> I think we need to evaluate the current schema file format from this OTEP's perspective before we mark format Stable.
> 
> My strong preference would be that any changes to the Schema file format that come from the OTEP are done in backward compatible way, by introducing new format versions. At the moment I don't see why it should not be possible.
> 
> @lquerel @jmacd @jsuereth any thoughts?

I do agree. With the modifications we plan to make to the existing Telemetry Schema format, we will need to introduce new format versions. It will also be beneficial to introduce a JSON Schema to facilitate the validation of this new format, as JSON Schema can validate both YAML and JSON. The same JSON Schema will also be useful for generating code to assist various telemetry schema consumers in easily parsing these files.

File: specification/schemas/file_format_v1.0.0.md
Comment (pellared) [inline specification/schemas/file_format_v1.0.0.md:1]: Shouldn't we move the "Schema File Format Number" sections to `README.md`? Version number handling should be something "generic".

Comment (MrAlias) [reply specification/schemas/file_format_v1.0.0.md:1]: Given this has been defined this way for 2 year, I don't see an issue.

This PR is scoped to stabilizing a part of the specification that has de facto become stable. I do not wish to introduce editorial changes to it. If in a future release we want to make an editorial change to the structure of the specification we could unify definitions then.

Comment (pellared) [reply specification/schemas/file_format_v1.0.0.md:1]: SMGT

Comment (pellared) [reply specification/schemas/file_format_v1.0.0.md:1]: SGMT. I created https://github.com/open-telemetry/opentelemetry-specification/issues/3851

**Review by pellared — APPROVED** [review]

**Comment by tigrannajaryan** [comment]
@lquerel this is another topic that the Schema workgroup will need to figure out.

@MrAlias I assume you do not need this resolved urgently, right? Once https://github.com/open-telemetry/oteps/pull/243 is merged there will be a workgroup that can deal with this.

**Comment by MrAlias** [comment]
I am hoping to have this merged sooner rather than latter. The Go SIG has many users that would like resource translations done for them but we cannot implement a solution in our stable package without this stabilizing.

I'm still failing to see where the conflict is coming from in stabilizing these two file formats.

If there are changes coming down the pipeline I expect they will be made in new file formats, not the existing ones. I would even go as far as saying that if someone recommended we change either of these file formats it should be blocked as it will break all of [these](https://github.com/open-telemetry/opentelemetry-specification/tree/main/schemas).

**Comment by MrAlias** [comment]
> I'm still failing to see where the conflict is coming from in stabilizing these two file formats.
> 
> If there are changes coming down the pipeline I expect they will be made in new file formats, not the existing ones. I would even go as far as saying that if someone recommended we change either of these file formats it should be blocked as it will break all of [these](https://github.com/open-telemetry/opentelemetry-specification/tree/main/schemas).

@lquerel @tigrannajaryan: can you help me understand where the reluctance to approve this comes from?

**Comment by MrAlias** [comment]
Rereading the [resource specification](https://github.com/open-telemetry/opentelemetry-specification/blob/a6ca2fd484c9e76fe1d8e1c79c99f08f4745b5ee/specification/resource/sdk.md#merge):

> The resulting resource MUST have all attributes that are on any of the two input resources. If a key exists on both the old and updating resource, the value of the updating resource MUST be picked (even if the updated value is empty).

Proves our aspiration to provide a translation in the Go SIG frivolous.

Closing as this is no longer a pressing matter.

**Pull request closed without merging**